js/ui/grid_core/ui.grid_core.keyboard_navigation.js

```diff
@@ -1998,13 +2003,15 @@ export default {
                 renderDelayedTemplates: function(change) {
                     this.callBase.apply(this, arguments);
                     if(!change || !change.repaintChangesOnly) {
-                        this.renderFocusState();
+                        const preventScroll = shouldPreventScroll(this);
```

Comment:
Maybe move this logic inside renderFocusState? Something like this:
```
renderFocusState: function(preventScroll) {
  const keyboardController = this.getController('keyboardNavigation');
  preventScroll = preventScroll ?? keyboardController.shouldPreventScroll();
  // with ES2021 can be changed to:
  // preventScroll ??= keyboardController.shouldPreventScroll();
  . . .
}
```
If we will need to render focus state without scroll preventing, we can pass false and it will work correctly. No argument - ask keyboardController for shouldPreventScroll. True - just prevent scroll.

Reply:
Technically, the scroll itself is implemented in FocusController._setFocusedRowElementTabIndex method. And the preventScroll parameter should be recalculated there. Since the **renderFocusState** method accepts the **preventScroll** parameter, I think it is better to calculate it before passing it to **renderFocusState**.